src/main/java/com/sandvoxel/generitech/common/tileentities/machines/TileEntityPulverizer.java

```diff
@@ -207,15 +205,39 @@ public long getPower() {
 
 
     @Override
-    public void update() {
-
-        //container.givePower(1000,false);
-
+    public void update()
+    {
 
+//      checking for the machine type
         if (machineTier == null)
             machineTier = MachineTier.byMeta(getBlockMetadata());
 
-        if (fuelRemaining == 0 && inventory.getStackInSlot(4) != null && net.minecraft.tileentity.TileEntityFurnace.getItemBurnTime(inventory.getStackInSlot(4)) > 0 && machineTier == MachineTier.TIER_0) {
+
+        if (container.getStoredPower() <= 0 && machineTier != MachineTier.TIER_0 && machineActive != false || fuelRemaining == 0 && machineTier == MachineTier.TIER_0 && machineActive != false )
+        {
+                machineActive = false;
+                this.markForUpdate();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "update" is 43 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)